src/charm.py

```diff
@@ -20,6 +21,7 @@
 
 from charms.nginx_ingress_integrator.v0.ingress import IngressRequires
 from leadership import LeadershipSettings
+from exceptions import *
```

Comment:
There's a `__all__` in exception library so the `import *` is controlled here.

Reply:
Yes I saw the `__all__` definition, it is still generally better to import the specific classes and functions rather than everything since it isn't clear from this whether there might be any name clashes with other imports